depthmapXcli/runmethods.cpp

```diff
@@ -257,6 +257,63 @@ namespace dm_runmethods
 
     }
 
+    void runSegmentAnalysis(const CommandLineParser &clp, const SegmentParser &sp, IPerformanceSink &perfWriter)
+    {
+        auto mGraph = loadGraph(clp.getFileName().c_str(), perfWriter);
+
+        auto state = mGraph->getState();
+
+        std::cout << "Running segment analysis... " << std::flush;
+        Options options;
+        options.radius_list = genshim::toPVector(sp.getRadii());
+        options.choice = sp.includeChoice();
+        options.tulip_bins = sp.getTulipBins();
+        switch(sp.getRadiusType()) {
+            case SegmentParser::RadiusType::SEGMENT_STEPS: {
+                options.radius_type = Options::RADIUS_STEPS;
+                break;
+            }
+            case SegmentParser::RadiusType::METRIC: {
+                options.radius_type = Options::RADIUS_METRIC;
+                break;
+            }
+            case SegmentParser::RadiusType::ANGULAR: {
+                options.radius_type = Options::RADIUS_ANGULAR;
+                break;
+            }
+            case SegmentParser::RadiusType::NONE:
+                break;
+        }
+        switch(sp.getAnalysisType()) {
```

Comment:
Not sure I understand this correctly? The `sp.getAnalysisType()` result seems to be only used in this switch statement, but `ANGULAR_FULL` and `ANGULAR_TULIP` result in the same call. So I guess the difference is that different parameters have been set to `options` before, and the parser checks that the parameters match chosen analysis type?
This is a bit confusing at best and hard to reason about in terms of correctness.

Reply:
Indeed, the MetaGraph checks whether the tulip bins are set to 0 which is not very clear. Addressed in commit 10a6a18